src/NServiceBus.Gateway/Utils/Guard.cs

```diff
@@ -0,0 +1,93 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Collections;
+    using System.Linq;
+    using System.Reflection;
+
+    static class Guard
+    {
+// ReSharper disable UnusedParameter.Global
+        public static void TypeHasDefaultConstructor(Type type, string argumentName)
+        {
+            if (type.GetConstructors(BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic)
+                .All(ctor => ctor.GetParameters().Length != 0))
```

Comment:
`ctor.GetParameters().Length != 0)` doesnt this assert that all constructors have zero params? not that one does?

Reply:
Removed the method since we don't use it but since it was copied from the core there might be a problem there?